### PR TITLE
bug(amplify-cli):resource-test file location issue resolved

### PR DESCRIPTION
**Pull request by studpeps**
#### Description of changes
Solved failing resource-test case issues by making all function input parameter file locations OS-specific

#### Issue #, if available
#7711 

#### Description of how you validated changes
1. cd packages/amplify-cli
2. yarn test 

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { stateManager, exitOnNextTick, ResourceDoesNotExistError, MissingParamet
 import * as inquirer from 'inquirer';
 import { updateBackendConfigAfterResourceRemove } from '../../../extensions/amplify-helpers/update-backend-config';
 import { removeResourceParameters } from '../../../extensions/amplify-helpers/envResourceParams';
-
+const path = require('path');
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts:6]: Can we move this to the top of the file and use `import` like:

```js
import * as path from 'path';
// or
import { join } from 'path';
```

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts [hunk 2]
```diff
@@ -215,7 +215,7 @@ describe('remove-resource', () => {
           },
         },
       });
-      expect(context.filesystem.remove).toBeCalledWith('backendDirPath/function/lambda1');
+      expect(context.filesystem.remove).toBeCalledWith(path.join("backendDirPath", "function","lambda1"));
```
Comment (cjihrig) [inline packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts:218]: Please use single quotes and add a space after the last comma.

**Review by cjihrig — COMMENTED** [review]
A couple of small comments.

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts (hunk 1 above)
Comment (studpeps) [reply packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts]: Surely

File: packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts (hunk 2 above)
Comment (studpeps) [reply packages/amplify-cli/src/__tests__/extensions/amplify-helpers/remove-resource.test.ts]: Committed these changes

**Review by cjihrig — APPROVED** [review]
LGTM, thank you!

**Pull request merged**